app/validators/answers/option_answer_validator.py

```diff
@@ -68,3 +102,38 @@ def validate_default_exists_in_options(self):
             option["value"] for option in self.options
         ]:
             self.add_error(self.ANSWER_DEFAULT_MISSING, default_value=default_value)
+
+    def validate_dynamic_options(self):
+        if not self.dynamic_options:
+            return None
+
+        self._validate_dynamic_options_answer_source()
+        self._validate_dynamic_options_value_rules()
+
+    def _validate_dynamic_options_value_rules(self):
+        for key_to_validate, allow_self_reference in [
+            ("values", False),
+            ("transform", True),
+        ]:
+            validator = RulesValidator(
+                self.dynamic_options[key_to_validate],
+                self.answer_id,
+                self.questionnaire_schema,
+                allow_self_reference=allow_self_reference,
+            )
+            self.errors += validator.validate()
+
+    def _validate_dynamic_options_answer_source(self):
+        if "source" not in self.dynamic_options["values"]:
+            return None
```

Comment:
same as above

Reply:
See above + here, I would mind doing the if check outside the method and calling the method when appropriate. Thoughts?.